src/test/run-pass/generator/static_generators.rs

```diff
@@ -0,0 +1,19 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
Can you add a test that static generators actually work - that you can yield `*b` and that it stays valid?

Reply:
You could do `yield *b` in movable generator, given NLL. Did you mean I should use the reference after the `yield` so the test still works with NLL?

Reply:
That's one thing. Another thing is that you should have a test that actually runs the generator and asserts that the references have the same value between resumes, to quickly catch if some change horribly breaks immovable generator codegen.

Reply:
I changed this test.